sdk/keyvault/azure-security-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/cryptography/CryptographyAsyncClient.java

```diff
@@ -606,18 +625,20 @@ public Mono<DecryptResult> decrypt(DecryptParameters decryptParameters) {
     @ServiceMethod(returns = ReturnType.SINGLE)
     public Mono<SignResult> sign(SignatureAlgorithm algorithm, byte[] digest) {
         try {
-            return withContext(context -> isValidKeyLocallyAvailable().flatMap(available -> {
-                if (!available) {
-                    return implClient.signAsync(algorithm, digest, context);
-                }
+            return withContext(context -> {
+                if (localKeyCryptographyClient != null) {
+                    if (!checkKeyPermissions(this.jsonWebKey.getKeyOps(), KeyOperation.SIGN)) {
+                        return Mono.error(LOGGER.logExceptionAsError(
+                            new UnsupportedOperationException(String.format(
+                                "The sign operation is not allowed for key with id: %s",
+                                this.jsonWebKey.getId()))));
+                    }
```

Comment:
This code is repeated in multiple places for different operations. We could extract this into a method to reduce duplication.